internal/hcs/process.go

```diff
@@ -163,7 +163,35 @@ func (process *Process) Kill(ctx context.Context) (bool, error) {
 		return true, nil
 	}
 
-	resultJSON, err := vmcompute.HcsTerminateProcess(ctx, process.handle)
+	// HCS serializes the signals sent to a target pid per compute system handle.
+	// To avoid SIGKILL being serialized behind other signals, we open a new compute
+	// system handle to deliver the kill signal.
+	// If the calls to opening a new compute system handle fail, we forcefully
+	// terminate the container itself so that no container is left behind
+	hcsSystem, err := OpenComputeSystem(ctx, process.system.id)
+	if err != nil {
+		// log error and force termination of container
+		log.G(ctx).WithField("err", err).Error("OpenComputeSystem() call failed")
+		err = process.system.Terminate(ctx)
+		// if the Terminate() call itself ever failed, log and return error
+		if err != nil {
+			log.G(ctx).WithField("err", err).Error("Terminate() call failed")
+			return false, err
+		}
+		process.system.Close()
+		return true, nil
```

Comment:
Are we assuming that if `OpenComputeSystem` or `OpenProcess` fail that it's always because the compute system/process does not exist anymore? Is there an error code we can check against, and return `false` otherwise? We probably shouldn't claim we delivered the signal if we failed for a different reason.

Reply:
Hi Kevin, Yes I made this assumption based one one comment of Kris "While OpenComputeSystem and OpenProcess can fail, the most likely failure is because the system or process no longer exists. Memory pressure could also cause failures but I would expect other things to be failing if there is that much memory pressure." . I am not sure if OpenComputeSystem() returns specific errors to help differentiate these cases from the other.

Reply:
This was marked resolved, what was the conclusion?